### PR TITLE
Converting a couple of version checks to modules detection in ESRestTestCase

### PR DESCRIPTION
**Pull request by ldematte**
Getting rid of a couple of version checks that are not needed.
This is preliminary work to make moving to features a bit easier (less features to introduce).

Assertions on Version are there to validate the change - will be removed with the other version checks.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -962,14 +967,23 @@ private Set<String> getAllUnexpectedTemplates() throws IOException {
...
+
         Request getShutdownStatus = new Request("GET", "_nodes/shutdown");
         Map<String, Object> statusResponse = responseAsMap(adminClient().performRequest(getShutdownStatus));
-        List<Map<String, Object>> nodesArray = (List<Map<String, Object>>) statusResponse.get("nodes");
-        List<String> nodeIds = nodesArray.stream().map(nodeShutdownMetadata -> (String) nodeShutdownMetadata.get("node_id")).toList();
+
+        Object nodesResponse = statusResponse.get("nodes");
+        final List<String> nodeIds;
+        if (nodesResponse instanceof List<?>) { // `nodes` is parsed as a List<> only if it's populated (not empty)
+            assert minimumNodeVersion().onOrAfter(Version.V_7_15_0);
```
Comment (rjernst) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:980]: Shouldn't this be a feature check?

**Review by rjernst — COMMENTED** [review]
This looks fine, but I think the title and description should be more focused, it is currently very generic. Perhaps something about fixing and converting some feature uses in ESRestTestCase?

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java (hunk 1 above)
Comment (ldematte) [reply test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:980]: This assert is here just to double-check the behaviour is not changed. I will remove this assert with the other version usage when we will have feature checks in rest tests.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by ldematte** [comment]
Thanks Ryan!

**Pull request merged**